### PR TITLE
Merge documentation process with the development process

### PR DESCRIPTION
**Pull request by pietroalbini**
The only two differences between the development process and the documentation process were:

* The requirement to open a GitHub issue for every change. Right now that is not a requirement in the development process, but we are going to change that in a future PR, and from that point onwards the process for docs and code changes should be the same.

* The requirement for the verification engineer to review the changes for compliance. This has been moved to the validation process, mentioning that the safety manager or a delegate must do that check.

### CODE REVIEW

File: ferrocene/doc/qualification-plan/src/development.rst [hunk 1]
```diff
@@ -9,10 +9,11 @@ Development Process
 The Ferrocene toolchain is a complex codebase that contains inputs from
 different teams and upstream sources. Ferrocene developers need to ensure that
 all changes made to it are properly reviewed, are of adequate quality, and can
-be properly assessed through automated testing. This section describes the
-process for merging any change into the Ferrocene codebase while accomplishing
-the aforementioned goals.
+be properly assessed through automated testing.
 
+This section describes the process for merging any change into the Ferrocene
+codebase while accomplishing the aforementioned goals. It appllies to both code
```
Comment (Veykril) [inline ferrocene/doc/qualification-plan/src/development.rst:15]:
```suggestion
codebase while accomplishing the aforementioned goals. It applies to both code
```

**Review by Veykril — APPROVED** [review]

**Comment by Veykril** [comment]
bors merge

**Review by tshepang — APPROVED** [review]
bors merge

**Comment by pietroalbini** [comment]
Rebased and fixed the failure.

**Review by tshepang — APPROVED** [review]
bors merge

**Pull request merged**